packages/common-scripts/src/type.ts

```diff
@@ -0,0 +1,27 @@
+import {
+  Cell,
+  CellCollector,
+  CellProvider,
+  QueryOptions,
+  Script,
+} from "@ckb-lumos/base";
+import { Options } from "@ckb-lumos/helpers";
+import { FromInfo } from "./from_info";
+
+export interface CellCollectorConstructor {
+  new (
+    fromInfo: FromInfo,
+    cellProvider: CellProvider,
+    {
+      config,
+      queryOptions,
+    }: Options & {
+      queryOptions?: QueryOptions;
+    }
+  ): CellCollectorType;
+}
+
+export interface CellCollectorType extends CellCollector {
+  readonly fromScript: Script;
+  collect(): AsyncGenerator<Cell>;
+}
```

Comment:
why are constructor and instance defined separately?

Reply:
`CellCollectorConstructor` for class type as` CellCollector: CellCollectorConstructor`,and `CellCollectorType` for instance side.
maybe I should change `CellCollectorType` to `CellCollectorInterface`